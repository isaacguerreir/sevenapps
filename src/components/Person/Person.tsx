import  * as React from 'react'
import { User } from '../../interfaces/User.interface'
import Grid from '@mui/material/Grid';
import Card from '@mui/material/Card';
import Typography from '@mui/material/Typography';
import CardContent from '@mui/material/CardContent';
import { styled } from '@mui/material/styles';

interface Props {
	key?: any
}
type PersonProps = Props & User

interface DataAssembled {
	Name: string,
	Age: number
}

const checkIfValid = (condition : boolean, value : any, outOfCase : any) => {
	if (!condition) console.warn("The props for Person are incorrect.") 
	return condition ? value : outOfCase
}

const validateProps = ({name, age} : PersonProps) : DataAssembled => {
	const Name : string = checkIfValid(name && name.length > 0, name, '')
	const Age : number = checkIfValid(age >= 0, age, 0)
	
	return { Name, Age }
}

const CustomCard = styled(Card)({
	margin: '1rem 0'
})

const Person = (props: PersonProps) => {
	const { Name, Age } = validateProps(props);
	return (
		<Grid xs={12} md={12}>
			<CustomCard variant="outlined">
					<CardContent>
						<Typography variant="h5" component="div">Name: {Name}</Typography>
						<Typography variant="h5" component="div">Age: {Name}</Typography>
					</CardContent>
			</CustomCard>
		</Grid>
	)	
}

export default Person
