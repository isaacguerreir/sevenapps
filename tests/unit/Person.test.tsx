import * as React from 'react'
import {cleanup, fireEvent, render} from '@testing-library/react';
import * as renderer from 'react-test-renderer';


import Person from '../../src/components/Person/Person'

describe("Person function component", () => {
	test("Should render person with name Roberto and age 31", async () => {
		const { getByText } = render(<Person name="Roberto" age="31"/>)
		
		expect(getByText('Name: Roberto')).toBeInTheDocument()
		expect(getByText('Age: 31')).toBeInTheDocument()
	})

	test("Should render person with name Isaac and age 0", async () => {
		const { getByText } = render(<Person name="Isaac" age="-1"/>)
		
		expect(getByText('Name: Isaac')).toBeInTheDocument()
		expect(getByText('Age: 0')).toBeInTheDocument()
	})

	
	test("Should render person with name '' and age 0 if not argument is passed", async () => {
		const { getByText } = render(<Person name="" age=""/>)
		
		expect(getByText('Name:')).toBeInTheDocument()
		expect(getByText('Age: 0')).toBeInTheDocument()
	})

	
	test("Should it render comparing to snapshot", async () => {
		const tree = renderer.create(<Person name="Lucas" age="26"/>)
    			.toJSON();
  		expect(tree).toMatchSnapshot();	
	})
})
