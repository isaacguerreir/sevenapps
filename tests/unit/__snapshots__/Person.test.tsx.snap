// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person function component Should it render comparing to snapshot 1`] = `
<div>
  <div>
    Name: 
    Lucas
  </div>
  <div>
    Age: 
    26
  </div>
</div>
`;
